docs/writing-tests/accessibility-testing.mdx

```diff
@@ -116,6 +116,87 @@ Customize the a11y ruleset at the story level by updating your story to include
 
 {/* prettier-ignore-end */}
 
+## Integration with Addon Test
+
+The a11y addon is compatible with Storybook's newest addon, the [test addon](./test-addon.mdx). When you run component tests, the a11y addon can automatically check for accessibility issues for all of your stories in the background. If there are any violations, the test will fail, and you will see the results in the sidebar.
+
+### Automatic setup
+
+When you add the a11y addon via [add](../api/cli-options.mdx#add) command, it will automatically be integrated with the component testing feature. You don't need to do anything else to enable accessibility checks in your component tests.
+
+### Manual setup
+
+If you have already a11y addon installed and you have upgraded manually to Storybook 8.5 or later, you can enable the integration by adding the following configuration to your `.storybook/vitest.config.ts` file:
+
+```diff
+// .storybook/vitest.config.ts
++import * as a11yAddonAnnotations from '@storybook/addon-a11y/preview';
+
+const annotations = setProjectAnnotations([
+  previewAnnotations,
++ a11yAddonAnnotations,
+]);
+
+// Run Storybook's beforeAll hook
+beforeAll(annotations.beforeAll);
+```
+
+### Violation impact levels
+
+By default, the addon will consider all violations as errors. However, you can configure the addon to consider some violations as warnings instead of errors. This can be useful when `@storybook/addon-a11y` is used in combination with `@storybook/experimental-addon-test`. To do this, set the `warnings` parameter in the `a11y` object to an array of impact levels that should be considered as warnings.
+
+```js
+export default {
+  title: 'button',
+  parameters: {
+    a11y: {
+      /**
+       * @default [ ]
+       * @type: Array<'minor' | 'moderate' | 'serious' | 'critical'>
+       */
+      warnings: ['minor', 'moderate'],
+    },
+  },
+};
+```
+
+In this example, all violations with an impact level of `minor` or `moderate` will be considered as warnings. All other violations will be considered as errors. When running automated UI tests featured by Vitest, all violations with an impact level of `serious` or `critical` will now fail the test. This failure is reflected as an error in the sidebar or when running Vitest separately. `minor` and `moderate` violations will be reported as warnings but will not fail the test.
```

Comment:
@valentinpalkovic when you're able can you:
- Adjust the text to something as follows (the words are not precious)
```suggestion
### Override accessibility violation levels

By default, when the accessibility addon runs with the test addon enabled, it interprets all violations as errors. This means that if a story has a minor accessibility violation, the test will fail. However, you can override this behavior by setting the `warnings` parameter in the `a11y` configuration object to define an array of impact levels that should be considered warnings.

{/* prettier-ignore-start */}

<CodeSnippets path="storybook-test-override-warning-levels.md" />

{/* prettier-ignore-end */}

In the example above, we configured all the `minor` or `moderate` accessibility violations to be considered warnings. All other levels (i.e., `serious` or `critical`) will continue to be considered errors, fail the test, and report the results accordingly in the Storybook UI or Vitest.
```
- Create the snippet file and include the necessary variants (i.e., JavaScript, TypeScript and TypeScript 4.9) using the boilerplate code you provided
```js
export default {
  title: 'button',
  parameters: {
    a11y: {
      /**
       * @default [ ]
       * @type: Array<'minor' | 'moderate' | 'serious' | 'critical'>
       */
      warnings: ['minor', 'moderate'],
    },
  },
};
```